great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -257,7 +257,23 @@ def list_keys(self, prefix=()):
                     key_list.append(key)
 
         return key_list
+   
     
+    def crmdir(self, mroot, curpath):
+        try:
+            while not os.listdir(curpath) and os.path.exists(curpath):
+                d2=curpath
+                f2=os.path.dirname(d2)
+                if os.path.isdir(curpath):
+                    shutil.rmtree(curpath)
+                else:
+                    os.remove(curpath)
+                curpath=f2
+                self.crmdir(mroot,curpath)
```

Comment:
Is this recursive call necessary if you're already doing a `while` loop? Also, don't we need to do something with `mroot` at some point? I think you can nix this recursive call if you add one more condition to the `while` loop: `and curpath != mroot`.

Reply:
I do use the same loop to delete the root directory.

Reply:
For instance, the last call below is from the recursive call.
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static
crmdir delete
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static/styles
new curpath
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static
crmdir delete
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static/fonts/HKGrotesk
new curpath
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static/fonts
crmdir delete
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static/fonts
new curpath
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static
crmdir delete
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456/static
new curpath
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456
crmdir delete
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs/s456
new curpath
/Users/priyajoseph/ge0316/great_expectations/great_expectations/uncommitted/data_docs
Cleaned data docs